File: FE/src/api/shop-auth.ts
import { http } from '@/utils/http';

export type ShopAuthStatus = 'pending' | 'approved' | 'rejected';

export interface ShopAuthInfo {
  id: string;
  userId: string;
  shopName: string;
  shopDesc?: string;
  category: string;
  licenseImage: string;
  legalPerson: string;
  phone: string;
  address: string;
  status: ShopAuthStatus;
  rejectReason?: string;
  createdAt: string;
  updatedAt: string;
}

export interface ShopAuthParams {
  shopName: string;
  shopDesc?: string;
  category: string;
  licenseImage: string;
  legalPerson: string;
  phone: string;
  address: string;
}

export const shopAuthApi = {
  // 提交店铺认证
  submitAuth: (params: ShopAuthParams) => {
    return http<number>({
      url: '/shop-auth',
      method: 'POST',
      data: params,
      loading: true
    });
  },

  // 获取认证状态
  getAuthStatus: () => {
    return http<ShopAuthInfo | null>({
      url: '/shop-auth/status',
      method: 'GET'
    });
  }
};

File: BE/XinMall/src/main/java/com/example/xinmall/entity/system/ShopAuth.java
package com.example.xinmall.entity.system;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.example.xinmall.entity.system.enums.ShopAuthStatus;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 店铺认证实体
 */
@Data
@TableName("shop_auth")
public class ShopAuth {

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 店铺名称
     */
    private String shopName;

    /**
     * 店铺描述
     */
    private String shopDesc;

    /**
     * 经营类目
     */
    private String category;

    /**
     * 营业执照图片
     */
    private String licenseImage;

    /**
     * 法人姓名
     */
    private String legalPerson;

    /**
     * 联系电话
     */
    private String phone;

    /**
     * 经营地址
     */
    private String address;

    /**
     * 认证状态
     */
    private ShopAuthStatus status;

    /**
     * 拒绝原因
     */
    private String rejectReason;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
}
